build.sbt

```diff
@@ -657,6 +658,32 @@ lazy val `text-buffer` = project
     )
   )
 
+val generateRustParserLib = TaskKey[Unit]("generateRustParserLib", "Generates parser native library")
+val generateRustParserLibSettings = generateRustParserLib := {
+    import sys.process._
+    Seq("cargo", "build", "-p", "enso-parser-jni") !
+}
+
+val generateRustParser = TaskKey[Unit]("generateRustParser", "Generates parser sources")
+val generateRustParserSettings = generateRustParser := {
+    import sys.process._
+    Seq("cargo", "run", "-p", "enso-parser-generate-java", "--bin", "enso-parser-generate-java", "lib/rust/parser/generate-java/java/org/enso/syntax2/") !
+}
+
+lazy val `syntax-rust-definition` = project
+  .in(file("lib/rust/parser"))
+  .configs(Test)
+  .settings(
+    compile := ((Compile / compile) dependsOn generateRustParser).value,
+    compile := ((Compile / compile) dependsOn generateRustParserLib).value,
+    Compile / javaSource := baseDirectory.value / "generate-java" / "java",
+    frgaalJavaCompilerSetting,
```

Comment:
Your insight is need, @hubertp.

 [LoadParser.sh](https://github.com/enso-org/enso/blob/79b58d1ee8f19656ec17850b270569124f0d8418/LoadParser.sh) shows the set of steps I need to execute to build the engine with proper lib `.so` file. Now I need to somehow hook that into the SBT chain.

If I invoke `sbt generateRustParser` and `generateRustParserLib`, then I get the right files (including `./target/rust/debug/libenso_parser.so`), but I am not sure how to set the dependency up properly?